### ISSUE TITLE
Note on fastmail configuration (url) misleading? 

### ISSUE THREAD

**Issue by bradyt** [untitaker/vdirsyncer#284]
https://vdirsyncer.readthedocs.org/en/stable/supported.html#fastmail suggests the abrupt url. 

    [storage card]
    type = carddav
    url = https://carddav.messagingengine.com/
    username = ...
    password = ...

I had an error until I used 

    [storage card]
    type = carddav
    url = https://carddav.messagingengine.com/dav/addressbooks/user/<username>/Default/
    username = ...
    password = ...

In other words, we find this at https://www.fastmail.com/help/technical/servernamesandports.html#contacts, 

> If your client does not support CardDAV auto-discovery, you may need to use a full URL path to your address book.

**Comment by untitaker** [comment]
Your CalDAV (calendar) URL points to an addressbook. This can't be right.

**Comment by untitaker** [comment]
The examples are correct. Could you show the full config?

**Comment by untitaker** [comment]
vdirsyncer does support autodiscovery though, I'd be interested in debug logs (vdirsyncer -vdebug discover)

On 26 October 2015 04:10:37 CET, bradyt <notifications@github.com> wrote:
>https://vdirsyncer.readthedocs.org/en/stable/supported.html#fastmail
>suggests the abrupt url. 
>
>    [storage cal]
>    type = caldav
>    url = https://caldav.messagingengine.com/
>    username = ...
>    password = ...
>
>I had an error until I used 
>
>    [storage cal]
>    type = caldav
>url =
>https://caldav.messagingengine.com/dav/addressbooks/user/<username>/Default/
>    username = ...
>    password = ...
>
>In other words, we find this at
>https://www.fastmail.com/help/technical/servernamesandports.html#contacts,
>
>
>> If your client does not support CardDAV auto-discovery, you may need
>to use a full URL path to your address book.
>
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/untitaker/vdirsyncer/issues/284

-- 
Sent from my phone. Please excuse my brevity.

**Comment by bradyt** [comment]
> Your CalDAV (calendar) URL points to an addressbook. This can't be right.

Sorry, `caldav` was copy paste typo. Corrected original post to `carddav`. Issue was/is (reproducible) with `carddav`.

**Comment by bradyt** [comment]
> The examples are correct. Could you show the full config?


    [general]
    status_path = ~/.vdirsyncer/status/
    
    [pair brady_contacts]
    a = brady_contacts_local
    b = brady_contacts_remote
    # collections = []
    # metadata = ["displayname"]
    
    [storage brady_contacts_local]
    type = filesystem
    path = ~/.contacts/
    fileext = .vcf
    
    [storage brady_contacts_remote]
    type = carddav
    # url = "https://carddav.messagingengine.com/dav/addressbooks/user/<my-handle>@fastmail.com/Default/"
    url = "https://carddav.messagingengine.com/"
    username = <my-handle>@fastmail.com
    password = ********


> vdirsyncer does support autodiscovery though, I'd be interested in debug logs (vdirsyncer -vdebug discover)

There's no error with `discover`; but with `sync`, 

    heeere:~ iam$ vdirsyncer -vdebug sync
    debug: Using 1 maximal workers.
    Syncing brady_contacts
    debug: Loaded status for brady_contacts
    debug: PROPFIND https://carddav.messagingengine.com/
    debug: {'Depth': 1, 'Content-Type': 'application/xml; charset=UTF-8', 'User-Agent': 'vdirsyncer'}
    debug: <?xml version="1.0" encoding="utf-8" ?>
    debug:             <D:propfind xmlns:D="DAV:">
    debug:                 <D:prop>
    debug:                     <D:resourcetype/>
    debug:                     <D:getcontenttype/>
    debug:                     <D:getetag/>
    debug:                 </D:prop>
    debug:             </D:propfind>
    debug:             
    debug: Sending request...
    debug: 404
    debug: {'Content-Encoding': 'gzip', 'Transfer-Encoding': 'chunked', 'Server': 'nginx', 'Connection': 'keep-alive', 'Date': 'Mon, 26 Oct 2015 18:27:57 GMT', 'Content-Type': 'text/html'}
    debug: <html>
    debug: <head><title>404 Not Found</title></head>
    debug: <body bgcolor="white">
    debug: <center><h1>404 Not Found</h1></center>
    debug: <hr><center>nginx</center>
    debug: </body>
    debug: </html>
    Unhandled exception occured for brady_contacts.
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/site-packages/vdirsyncer/cli/tasks.py", line 65, in sync_collection
        force_delete=force_delete
      File "/usr/local/lib/python2.7/site-packages/vdirsyncer/sync.py", line 163, in sync
        b_info.prepare_idents(storage_a.read_only)
      File "/usr/local/lib/python2.7/site-packages/vdirsyncer/sync.py", line 98, in prepare_idents
        for href, etag in self.storage.list():
      File "/usr/local/lib/python2.7/site-packages/vdirsyncer/storage/dav.py", line 569, in list
        headers=headers)
      File "/usr/local/lib/python2.7/site-packages/vdirsyncer/storage/dav.py", line 326, in request
        return utils.http.request(method, url, session=self._session, **more)
      File "/usr/local/lib/python2.7/site-packages/vdirsyncer/utils/http.py", line 96, in request
        raise exceptions.NotFoundError(r.reason)
    NotFoundError: Not Found

**Comment by bradyt** [comment]
However, I reconsidered your documentation. Your mention of `discover` made me reread things a little differently, so I note that adding `collections = ["from b"]` to `pair` section eliminates any error on `sync`. 

I will close this issue, thank you for considering my misstep.

**Comment by untitaker** [comment]
Ah, I guess you hadn't a collections param before? Yeah, a common issue, I still wonder about ways to improve UI...
